.eslintrc.js

```diff
@@ -0,0 +1,28 @@
+// It has ts files setup not applied for the whole project just for any js configuration files at the root
+// such as this file for instance.
+module.exports = {
+  extends: ['eslint:recommended', 'prettier'],
+  plugins: ['prettier'],
+  env: {
+    node: true,
+    mocha: true,
+  },
+  overrides: [
```

Comment:
Why are use an override? All files we want checked are in typescript...?

Reply:
There is a comment in the first line of this file explaining the reason for that. But I just noticed it is not well written sorry. The reason is because of configuration files like `.eslintrc.js` itself are `.js` and not `.ts` and it was failing the linter. So, I applied the typescript settings only for folders we know must contain only typescript files. But I guess I could just ignore `.eslintrc.js` and leave the typescript settings applied to the whole project.

Reply:
Could we then just define this file as a `.eslintrc.json` again instead? Do we need this to be a js file?

Reply:
Using js instead of json is nice because it allows comments. But at least for now we don't have any complex setup which really requires comments, so should be fine going back to json.